### ISSUE TITLE
NTP.now() and NTP.getNtpOffset() taking forever to return value. 

### ISSUE THREAD

**Issue by fonkamloic** [knezzz/ntp#21]
Both functions take forever to execute. 

Trying NTP.now().then() yields nothing meanwhile, using an await puts my app in a deadlock. 

I'm currently using Flutter 1.20.2 and Dart 2.9.1 on a MacOX

**Comment by rajveermalviya** [comment]
you can try changing `lookUpAddress`

default is `pool.ntp.org`

alternatives are: 
- `time.google.com`
- `time.cloudflare.com`


```dart
await NTP.getNtpOffset(lookUpAddress: 'time.cloudflare.com');
```

**Comment by fonkamloic** [comment]
Thanks a ton.

**Comment by dokenedgar** [comment]
@rajveermalviya Did what you recommended but still the NTP.now() methods takes forever.
Was there anything you did additionally to fix your code @fonkamloic ?

**Comment by rajveermalviya** [comment]
@dokenedgar `NTP.now()` doesn't support setting custom `lookUpAddress`.

you can create following helper to use custom `lookUpAddress`.

```dart
Future<DateTime> ntpNow({lookUpAddress = 'time.cloudflare.com'}) async {
  final localTime = DateTime.now();

  final offset = await NTP.getNtpOffset(
    lookUpAddress: lookUpAddress,
    localTime: localTime,
  );

  return localTime.add(Duration(milliseconds: offset));
}
```

To change the lookUpAddress
```dart
final networkTime = await ntpNow(lookUpAddress: 'time.google.com');
```

**Comment by dokenedgar** [comment]
Thanks for your response @rajveermalviya . Tried it the way you wrote and it worked. However, out of curiosity I tried loading the time using the initial `pool.ntp.org` and it worked, this made me very curious: is it that the pool.ntp.org is experiencing outages of recent? And if you can, which one is more reliable as I have an app in production for a client and getting the network time is crucial to how the app works. Should I stick with the wrapper, implying it's more reliable or be taking a risk with the `pool.ntp.org`?
Or maybe have a timer that loops through the 3 addresses after certain interval?

**Comment by rajveermalviya** [comment]
`pool.ntp.org` is a pool of NTP servers, on doing a DNS lookup it returns the nearest NTP server's IP for your IP address zone.

For me, it sometimes returns IP of `time.cloudflare.com` by default, (location is India).
You can check what server it connects to and also server's response times, for your preference by running
```bash
ping pool.ntp.org
```
run it many times and check if `time.cloudflare.com` comes up.
`time.google.com` is used by all Android devices by default (at the time of device setup).

PS. I would recommend doing the test and finding the best server, rather than using `pool.ntp.org` because it adds unnecessary overhead (for finding the nearest server). If your client is not based in a single country then go for `time.google.com` or `time.cloudflare.com` because they are the most reliable.

**Comment by rajveermalviya** [comment]
@dokenedgar 
Find more info about pool.ntp.org project here https://www.ntppool.org/en/

**Comment by dokenedgar** [comment]
Thank you so much @rajveermalviya  I understand the whole concept better now. And the ping tests seem to be showing responses from time.cloudfare.com.
Thanks again.

**Comment by knezzz** [comment]
I merged PR that adds timeout as well, so that sould help.
I will leave this issue open until I get time to add this to the documentation 😃 or add something to have more options to choose lookup address even for `NTP.now()`

**Comment by knezzz** [comment]
Merged PR where `time.google.com` is default lookup, and you can cnage lookup address when using `.now()`
Available in 1.0.8

Closing this issue now

**Comment by iamhx** [comment]
One of my user has encountered a problem with my app loading forever and I believe this issue could have been the problem, as all my own APIs have timeouts.

They have mentioned that they were not able to load my app when using the WiFi of their workplace, but otherwise on cellular data and home WiFi everything is normal.

My question is, what seems to be causing this issue such that certain networks blocks this? What is the best solution to solve this too? Timeouts wouldn't seem to help if the default lookup address is being blocked.
